CryptoPkg/Library/TlsLibMbedtls/TlsConfig.c

```diff
@@ -631,13 +636,14 @@ TlsSetCaCertificate (
     return EFI_INVALID_PARAMETER;
   }
 
-  mbedtls_x509_crt_init(&Crt);
+  Crt = AllocateZeroPool(sizeof(mbedtls_x509_crt));
+  mbedtls_x509_crt_init(Crt);
 
-  Ret = mbedtls_x509_crt_parse_der(&Crt, Data, DataSize);
+  Ret = mbedtls_x509_crt_parse_der(Crt, Data, DataSize);
 
   if (Ret == 0) {
-    mbedtls_ssl_conf_ca_chain((mbedtls_ssl_config *)TlsConn->Ssl->conf, &Crt, NULL);
-    mbedtls_x509_crt_free(&Crt);
+    mbedtls_ssl_conf_ca_chain((mbedtls_ssl_config *)TlsConn->Ssl->conf, Crt, NULL);
+    mbedtls_ssl_conf_cert_profile((mbedtls_ssl_config *)TlsConn->Ssl->conf, &mbedtls_x509_crt_profile_default);

```

Comment:
can we move mbedtls_ssl_conf_cert_profile and mbedtls_ssl_conf_sig_algs to init function?